### ISSUE TITLE
Add customization to the inference

### ISSUE THREAD

**Issue by jsancs** [axolotl-ai-cloud/axolotl#505]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Allow to configure the inference parameters from the yaml configuration file, just as it is done with the training parameters, to have more customization in the inference phase.

At this part of the _do_inference()_ method:
```
print("=" * 40)
model.eval()
with torch.no_grad():
      generation_config = GenerationConfig(
      repetition_penalty=1.1,
      max_new_tokens=1024,
      temperature=0.9,
      top_p=0.95,
      top_k=40,
      bos_token_id=tokenizer.bos_token_id,
      eos_token_id=tokenizer.eos_token_id,
      pad_token_id=tokenizer.pad_token_id,
      do_sample=True,
      use_cache=True,
      return_dict_in_generate=True,
      output_attentions=False,
      output_hidden_states=False,
      output_scores=False,
)
streamer = TextStreamer(tokenizer)
generated = model.generate(
      inputs=batch["input_ids"].to(cfg.device),
      generation_config=generation_config,
      streamer=streamer,
)
print("=" * 40)
```

### ✔️ Solution

My proposed solution is to allow configuring these parameters from the inference section in the yaml file.

For example:

```
inference:
      temperature: 1
      top_k: 30
      top_p: 90
```
and so on with all the parameters you want to configure, keeping a default value for those that do not.

### ❓ Alternatives

Any other solution that allows customization of these parameters would also be satisfactory.

### 📝 Additional Context

I could work on this feature. I just need feedback and approval.

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by mhenrichsen** [comment]
Thanks for the suggestion. If you want to work on the feature, be very welcome :)

Going to mark as stale for now.
